kiwi/storage/subformat/vagrant_libvirt.py

```diff
@@ -15,63 +15,26 @@
 # You should have received a copy of the GNU General Public License
 # along with kiwi.  If not, see <http://www.gnu.org/licenses/>
 #
-import json
 import os
-import random
+from typing import Dict  # pylint: disable=unused-import
 from textwrap import dedent
-from tempfile import mkdtemp
 
 # project
-from kiwi.storage.subformat.base import DiskFormatBase
+from kiwi.storage.subformat.vagrant_base import DiskFormatVagrantBase
 from kiwi.storage.subformat.qcow2 import DiskFormatQcow2
 from kiwi.command import Command
 
-from kiwi.exceptions import (
-    KiwiFormatSetupError
-)
 
-
-class DiskFormatVagrantLibVirt(DiskFormatBase):
+class DiskFormatVagrantLibVirt(DiskFormatVagrantBase):
+    provider = 'libvirt'
```

Comment:
I think I overlooked something, but where is this variable used ?

Reply:
This is used in `DiskFormatVagrantBase.post_init` where the base class constructs the image format from the provider name.

Reply:
Ah ok now I saw it

ok I was looking for where self.provider gets initialized. I think DiskFormatVagrantBase should initialize a default provider. The setting for ```self.image_format = 'vagrant.' + self.provider + '.box'``` is provider specific and imho doesn't belong to the base class. I think DiskFormatVagrantBase should call another post_init() such that the specialized provider class can overload that method and just set ```self.provider = 'libvirt'```. The way you did it does not loosely couple the classes and ties them hard together. The methods in the base class that references ```self.image_format``` should only reference self.provider with the value that it has at call time or its default value.

Sorry if that sounds nitpicking but it took me a while to find out how provider is used as it's class global. We should avoid that imho